clash-cores/src/Clash/Cores/Xilinx/Xpm/Cdc/ArraySingle.hs

```diff
@@ -0,0 +1,90 @@
+{-|
+  Copyright   :  (C) 2023, Google LLC
+  License     :  BSD2 (see the file LICENSE)
+  Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Clash.Cores.Xilinx.Xpm.Cdc.ArraySingle
+  ( xpmCdcArraySingle
+  , XpmCdcArraySingleConfig(..)
+  , xpmCdcArraySingleWith
+  ) where
+
+import GHC.Stack (HasCallStack)
+
+import Clash.Explicit.Prelude
+
+import Clash.Cores.Xilinx.Xpm.Cdc.ArraySingle.Internal (xpmCdcArraySingle#)
+
+-- | Synchronizes an array of independent bits from the source clock domain to
+-- the destination. Be careful when using this primitive: each bit will be
+-- synchronized independently. For more information see [PG382](https://docs.xilinx.com/r/en-US/pg382-xpm-cdc-generator/XPM_CDC_ARRAY_SINGLE).
+--
+-- __N.B.__: In order to simulate initial values, both the source and destination
+--           domain need to support them. If the source and destination domain
+--           disagree on this property, use of this function will fail to
+--           simulate and translate to HDL. You can explicitly set it using
+--           'xpmCdcArraySingleWith'.
+xpmCdcArraySingle ::
+  forall n src dst.
+  ( 1 <= n, n <= 1024
+  , KnownDomain src
+  , KnownDomain dst
+  , KnownNat n
+  , HasCallStack
+  ) =>
+  Clock src ->
+  Clock dst ->
+  Signal src (BitVector n) ->
+  Signal dst (BitVector n)
```

Comment:
Will people ever _not_ use this as a `Vec _ a` where `BitSize a ~ 1`? Since the bits are independent, it seems like a better match.

Also, it seems helpful to do the pack/unpack as part of the provided function, so people don't have to do it themselves.

Reply:
Yeah good question, I'll just make it `a` / `BitPack a`.